Adding logic for get status command

nikhiljindal opened the pull request:
Adding a `get-status` command that prints information about the load balancer.
It prints the IP address and list of clusters that it is spread to.

Using ForwardingRule.Description to store this information.

cc @nicksardo @bowei @csbell @madhusudancs @G-Harmon

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/googlecloudplatform/k8s-multicluster-ingress/14)
<!-- Reviewable:end -->

G-Harmon commented:
lgtm after my comments.

---

Reviewed 13 of 13 files at r1.
Review status: all files reviewed at latest revision, 9 unresolved discussions.

---

*[app/mci/cmd/create_test.go, line 111 at r1](https://reviewable.io:443/reviews/googlecloudplatform/k8s-multicluster-ingress/14#-KxiD8CEsXuR3LL5twtr:-KxiD8CEsXuR3LL5twts:b-trl3kl) ([raw file](https://github.com/googlecloudplatform/k8s-multicluster-ingress/blob/bf31edd2914e4cd801c2db15b5d7887310331c7d/app/mci/cmd/create_test.go#L111)):*
> ```Go
> 
> 	runFn := func() error {
> 		_, err := createIngress("kubeconfig", "../../../testdata/ingress.yaml")
> ```

validate the list of clusters?

---

*[app/mci/cmd/getstatus.go, line 93 at r1](https://reviewable.io:443/reviews/googlecloudplatform/k8s-multicluster-ingress/14#-KxiDpXEus0j0SUoWECV:-KxiDpXEus0j0SUoWECW:b-7vwnwt) ([raw file](https://github.com/googlecloudplatform/k8s-multicluster-ingress/blob/bf31edd2914e4cd801c2db15b5d7887310331c7d/app/mci/cmd/getstatus.go#L93)):*
> nil
please include comment for literals (/*var name*/)

---

*[app/mci/pkg/gcp/forwardingrule/forwardingrulesyncer.go, line 161 at r1](https://reviewable.io:443/reviews/googlecloudplatform/k8s-multicluster-ingress/14#-KxiJvCf8szLvOmm6gOy:-KxiJvCf8szLvOmm6gOz:buhoqhd) ([raw file](https://github.com/googlecloudplatform/k8s-multicluster-ingress/blob/bf31edd2914e4cd801c2db15b5d7887310331c7d/app/mci/pkg/gcp/forwardingrule/forwardingrulesyncer.go#L161)):*
> ```Go
> 		IPAddress:        ipAddress,
> 	}
> 	desc, err := status.ToString()
> ```

nit: I think this might be easier to follow if you did s/desc/lbDesc/

---

*[app/mci/pkg/gcp/forwardingrule/forwardingrulesyncer_test.go, line 98 at r1](https://reviewable.io:443/reviews/googlecloudplatform/k8s-multicluster-ingress/14#-KxiKDp-7-p6B-u8T6xY:-KxiKDp-7-p6B-u8T6xZ:b-3tgdvm) ([raw file](https://github.com/googlecloudplatform/k8s-multicluster-ingress/blob/bf31edd2914e4cd801c2db15b5d7887310331c7d/app/mci/pkg/gcp/forwardingrule/forwardingrulesyncer_test.go#L98)):*
> 1"
Use different cluster names...

---

*[app/mci/pkg/gcp/forwardingrule/forwardingrulesyncer_test.go, line 112 at r1](https://reviewable.io:443/reviews/googlecloudplatform/k8s-multicluster-ingress/14#-KxiKN7xas2RtM37XM6u:-KxiKN7xas2RtM37XM6v:baqodwq) ([raw file](https://github.com/googlecloudplatform/k8s-multicluster-ingress/blob/bf31edd2914e4cd801c2db15b5d7887310331c7d/app/mci/pkg/gcp/forwardingrule/forwardingrulesyncer_test.go#L112)):*
> Fatalf
these two Fatals could be Errors

---

*[app/mci/pkg/gcp/forwardingrule/interfaces.go, line 24 at r1](https://reviewable.io:443/reviews/googlecloudplatform/k8s-multicluster-ingress/14#-KxiKXtg5hk8sN9aEh8J:-KxiKXth5Ooj4qae50T6:b-7x9rv8) ([raw file](https://github.com/googlecloudplatform/k8s-multicluster-ingress/blob/bf31edd2914e4cd801c2db15b5d7887310331c7d/app/mci/pkg/gcp/forwardingrule/interfaces.go#L24)):*
> This is used to generate status for the load balancer.
This is true, but maybe also add that the info is stored in the forwarding rule's description?

---

*[app/mci/pkg/gcp/loadbalancer/loadbalancersyncer.go, line 90 at r1](https://reviewable.io:443/reviews/googlecloudplatform/k8s-multicluster-ingress/14#-KxiKvcovPbrRLz6mJJt:-KxiKvcovPbrRLz6mJJu:bxpj45q) ([raw file](https://github.com/googlecloudplatform/k8s-multicluster-ingress/blob/bf31edd2914e4cd801c2db15b5d7887310331c7d/app/mci/pkg/gcp/loadbalancer/loadbalancersyncer.go#L90)):*
>  clusters []string
document the new parameter in the interface?

---

*[app/mci/pkg/gcp/loadbalancer/loadbalancersyncer.go, line 174 at r1](https://reviewable.io:443/reviews/googlecloudplatform/k8s-multicluster-ingress/14#-KxiLbtxMKgi5CO1U2bX:-KxiLbtxMKgi5CO1U2bY:b-5pgkbm) ([raw file](https://github.com/googlecloudplatform/k8s-multicluster-ingress/blob/bf31edd2914e4cd801c2db15b5d7887310331c7d/app/mci/pkg/gcp/loadbalancer/loadbalancersyncer.go#L174)):*
> ```Go
> 
> // PrintStatus prints the current status of the load balancer.
> func (l *LoadBalancerSyncer) PrintStatus() {
> ```

I'd suggest making this "GetStatus" and returning a string. That way the final consumer can decide how/when to print it to some output stream.

---

*[app/mci/pkg/gcp/status/loadbalancerstatus.go, line 10 at r1](https://reviewable.io:443/reviews/googlecloudplatform/k8s-multicluster-ingress/14#-KxiEmWW0a0ecHf9Khnm:-KxiEmWW0a0ecHf9Khnn:b-kj68xs) ([raw file](https://github.com/googlecloudplatform/k8s-multicluster-ingress/blob/bf31edd2914e4cd801c2db15b5d7887310331c7d/app/mci/pkg/gcp/status/loadbalancerstatus.go#L10)):*
> // Description about the resource this status description is on.
This comment is a bit clunky.  maybe something about how it's is a serialized form of this LoadBalancerStatus?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/googlecloudplatform/k8s-multicluster-ingress/14#-:-KxiM3SnpcOaUDsVLAF3:b2dik0h)*
<!-- Sent from Reviewable.io -->

nikhiljindal commented:
Review status: all files reviewed at latest revision, 9 unresolved discussions.

---

*[app/mci/cmd/create_test.go, line 111 at r1](https://reviewable.io:443/reviews/googlecloudplatform/k8s-multicluster-ingress/14#-KxiD8CEsXuR3LL5twtr:-KxiObuockF6ETqnHfmc:b-896fix) ([raw file](https://github.com/googlecloudplatform/k8s-multicluster-ingress/blob/bf31edd2914e4cd801c2db15b5d7887310331c7d/app/mci/cmd/create_test.go#L111)):*
<details><summary><i>Previously, G-Harmon wrote…</i></summary><blockquote>

validate the list of clusters?
</blockquote></details>

Done.

---

*[app/mci/cmd/getstatus.go, line 93 at r1](https://reviewable.io:443/reviews/googlecloudplatform/k8s-multicluster-ingress/14#-KxiDpXEus0j0SUoWECV:-KxiOqnJdvRrFi_AJh0r:b-896fix) ([raw file](https://github.com/googlecloudplatform/k8s-multicluster-ingress/blob/bf31edd2914e4cd801c2db15b5d7887310331c7d/app/mci/cmd/getstatus.go#L93)):*
<details><summary><i>Previously, G-Harmon wrote…</i></summary><blockquote>

> nil
please include comment for literals (/*var name*/)
</blockquote></details>

Done.

---

*[app/mci/pkg/gcp/forwardingrule/forwardingrulesyncer.go, line 161 at r1](https://reviewable.io:443/reviews/googlecloudplatform/k8s-multicluster-ingress/14#-KxiJvCf8szLvOmm6gOy:-KxiP4bLvvVG2j9ggvT3:b-36xc1y) ([raw file](https://github.com/googlecloudplatform/k8s-multicluster-ingress/blob/bf31edd2914e4cd801c2db15b5d7887310331c7d/app/mci/pkg/gcp/forwardingrule/forwardingrulesyncer.go#L161)):*
<details><summary><i>Previously, G-Harmon wrote…</i></summary><blockquote>

nit: I think this might be easier to follow if you did s/desc/lbDesc/
</blockquote></details>

To clarify, this is the string that goes in description field of forwarding rule, so desc or FrDesc is a better name than lbDesc?

---

*[app/mci/pkg/gcp/forwardingrule/forwardingrulesyncer_test.go, line 98 at r1](https://reviewable.io:443/reviews/googlecloudplatform/k8s-multicluster-ingress/14#-KxiKDp-7-p6B-u8T6xY:-KxiPOFMqvdJjQtzqE2P:b-6k4730) ([raw file](https://github.com/googlecloudplatform/k8s-multicluster-ingress/blob/bf31edd2914e4cd801c2db15b5d7887310331c7d/app/mci/pkg/gcp/forwardingrule/forwardingrulesyncer_test.go#L98)):*
<details><summary><i>Previously, G-Harmon wrote…</i></summary><blockquote>

> 1"
Use different cluster names...
</blockquote></details>

haha fixed :)

---

*[app/mci/pkg/gcp/forwardingrule/forwardingrulesyncer_test.go, line 112 at r1](https://reviewable.io:443/reviews/googlecloudplatform/k8s-multicluster-ingress/14#-KxiKN7xas2RtM37XM6u:-KxiPWP_KU2bCxAPzFSW:b-896fix) ([raw file](https://github.com/googlecloudplatform/k8s-multicluster-ingress/blob/bf31edd2914e4cd801c2db15b5d7887310331c7d/app/mci/pkg/gcp/forwardingrule/forwardingrulesyncer_test.go#L112)):*
<details><summary><i>Previously, G-Harmon wrote…</i></summary><blockquote>

> Fatalf
these two Fatals could be Errors
</blockquote></details>

Done.

---

*[app/mci/pkg/gcp/forwardingrule/interfaces.go, line 24 at r1](https://reviewable.io:443/reviews/googlecloudplatform/k8s-multicluster-ingress/14#-KxiKXtg5hk8sN9aEh8J:-KxiQ3VdWSmyMVQNbdJw:b-unxjc2) ([raw file](https://github.com/googlecloudplatform/k8s-multicluster-ingress/blob/bf31edd2914e4cd801c2db15b5d7887310331c7d/app/mci/pkg/gcp/forwardingrule/interfaces.go#L24)):*
<details><summary><i>Previously, G-Harmon wrote…</i></summary><blockquote>

> This is used to generate status for the load balancer.
This is true, but maybe also add that the info is stored in the forwarding rule's description?
</blockquote></details>

Thats is an implementation detail. Other implementations can store it somewhere else.
Added that comment to the implementation method.
Removed "This is used to generate status" line from here.

---

*[app/mci/pkg/gcp/loadbalancer/loadbalancersyncer.go, line 90 at r1](https://reviewable.io:443/reviews/googlecloudplatform/k8s-multicluster-ingress/14#-KxiKvcovPbrRLz6mJJt:-KxiQX2Z-R8lXpPtOmpT:b-896fix) ([raw file](https://github.com/googlecloudplatform/k8s-multicluster-ingress/blob/bf31edd2914e4cd801c2db15b5d7887310331c7d/app/mci/pkg/gcp/loadbalancer/loadbalancersyncer.go#L90)):*
<details><summary><i>Previously, G-Harmon wrote…</i></summary><blockquote>

>  clusters []string
document the new parameter in the interface?
</blockquote></details>

Done.

---

*[app/mci/pkg/gcp/loadbalancer/loadbalancersyncer.go, line 174 at r1](https://reviewable.io:443/reviews/googlecloudplatform/k8s-multicluster-ingress/14#-KxiLbtxMKgi5CO1U2bX:-KxiRtz6OJntbEQ4-QmJ:b-896fix) ([raw file](https://github.com/googlecloudplatform/k8s-multicluster-ingress/blob/bf31edd2914e4cd801c2db15b5d7887310331c7d/app/mci/pkg/gcp/loadbalancer/loadbalancersyncer.go#L174)):*
<details><summary><i>Previously, G-Harmon wrote…</i></summary><blockquote>

I'd suggest making this "GetStatus" and returning a string. That way the final consumer can decide how/when to print it to some output stream.
</blockquote></details>

Done.

---

*[app/mci/pkg/gcp/status/loadbalancerstatus.go, line 10 at r1](https://reviewable.io:443/reviews/googlecloudplatform/k8s-multicluster-ingress/14#-KxiEmWW0a0ecHf9Khnm:-KxiTAK2H1WUhi7kdtF5:b-bu91xm) ([raw file](https://github.com/googlecloudplatform/k8s-multicluster-ingress/blob/bf31edd2914e4cd801c2db15b5d7887310331c7d/app/mci/pkg/gcp/status/loadbalancerstatus.go#L10)):*
<details><summary><i>Previously, G-Harmon wrote…</i></summary><blockquote>

> // Description about the resource this status description is on.
This comment is a bit clunky.  maybe something about how it's is a serialized form of this LoadBalancerStatus?
</blockquote></details>

Updated

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/googlecloudplatform/k8s-multicluster-ingress/14)*
<!-- Sent from Reviewable.io -->
